doc/source/development/dev_documentation.rst

```diff
@@ -341,6 +341,46 @@ To reference a method or function::
   :cpp:func:`MyClass::MyMethod`
   :cpp:func:`MyFunction`
 
+
+Command-line program usage
+--------------------------
+
+To document a command-line tool, use the ``example`` directive.
+An example may have a title as well as an id that can be used to generate
+a cross-reference from the same document.
+
+.. code-block:: rst
+
+    .. example::
+       :title: Listing files
+       :id: `basic-ls`
```

Comment:
From the actual documentation pages, it doesn't look those backwards single quote are needed:
```suggestion
       :id: basic-ls
```